src/clients/FtpGateway.cc

```diff
@@ -532,7 +532,10 @@ ftpListParseParts(const char *buf, struct Ftp::GatewayFlags flags)
     ftpListParts *p = NULL;
     char *t = NULL;
     const char *ct = NULL;
-    char *tokens[MAX_TOKENS];
+    struct FtpLineToken {
```

Comment:
Deleting the default move constructor here produces compile errors, so I had to add a constructor. Adding a new constructor, the code did not looked very good, because I had the feeling that a destructor required and moreover  the code at the end of this method which detele one-by-one array elements did not looks very good.
I had to make many changes so I decided to not do anything.

Is it a huge problem? This is just a C code which survived since squid 1x and 2x days.

Reply:
> Is it a huge problem?

If there is copying/moving, it is a huge problem. Otherwise, it is not. What compiler errors do you get when you disable copying/moving?

Reply:
`FtpGateway.cc:539:24: error: no matching function for call to ‘ftpListParseParts(const char*, Ftp::GatewayFlags)::FtpLineToken::FtpLineToken()’
     } tokens[MAX_TOKENS];
`